### ISSUE TITLE
[tvplay] extractor deprecation

### ISSUE THREAD

**Issue by 18928172992817182** [rg3/youtube-dl#30084]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2021.06.06**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
gustaf@tuxbox-tuxnet youtube-dl$ youtube-dl -v https://www.viafree.fi/ohjelmat/entertainment/amazing-makeovers/kausi-7/jakso-2
[debug] System config: ['--prefer-free-formats']
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.viafree.fi/ohjelmat/entertainment/amazing-makeovers/kausi-7/jakso-2']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, pref UTF-8
[debug] youtube-dl version 2021.06.06
[debug] Python version 3.9.7 (CPython) - Linux-5.14.9-200.fc34.x86_64-x86_64-with-glibc2.33
[debug] exe versions: ffmpeg 4.4, ffprobe 4.4
[debug] Proxy map: {}
[generic] jakso-2: Requesting header
WARNING: Falling back on generic information extractor.
[generic] jakso-2: Downloading webpage
[generic] jakso-2: Extracting information
ERROR: Unsupported URL: https://www.viafree.fi/ohjelmat/entertainment/amazing-makeovers/kausi-7/jakso-2
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/youtube_dl-2021.6.6-py3.9.egg/youtube_dl/YoutubeDL.py", line 815, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/youtube_dl-2021.6.6-py3.9.egg/youtube_dl/YoutubeDL.py", line 836, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/local/lib/python3.9/site-packages/youtube_dl-2021.6.6-py3.9.egg/youtube_dl/extractor/common.py", line 534, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/lib/python3.9/site-packages/youtube_dl-2021.6.6-py3.9.egg/youtube_dl/extractor/generic.py", line 3489, in _real_extract
    raise UnsupportedError(url)
youtube_dl.utils.UnsupportedError: Unsupported URL: https://www.viafree.fi/ohjelmat/entertainment/amazing-makeovers/kausi-7/jakso-2


```

## Description

Tvplay needs major refactoring.

This issue is related to #12685 and #28998. The extractor tvplay is severely deprecated. It was develop at a time when MTG was providing many different services which now have been sold or canceled. MTG itself has been renamed to NENT.

I have gone through this entire process already over at https://github.com/streamlink/streamlink/issues/4030.

However, there the solution was to drop support for it altogether since the live streaming had been canceled on the remaining MTG services called Viafree.

Viafree is now available in Sweden, Norway, Denmark and Finland. Before the restructuring of MTG they also provided services in Latvia, Estonia and Lithuania with a common backend. However these have been sold and are now handled by a different company which is the cause of #28998.

This issue is about refactoring or even removing the tvplay extractor and replacing it with the appropriate extractors. Tvplay is only working on Viafree now so my suggestion like in streamlink is that we refactor the tvplay extractor to just a viafree extractor because that is the only thing that remains and can be supported by the MTG API in the remains of tvplay.

In this process we can also add support for Finland because they were left out of tvplay for some reason.

All the other services from the Baltics that MTG sold that were part of the tvplay extractor need a completely new extractor in order to work because it is a different backend and service provider now. But all of them actually do share the same underlying API for the Baltics. It is just not provided by MTG/NENT. That implementation however should probably be done in #28998.

That is the full story. I can start to work on fixing tvplay.

**Comment by 18928172992817182** [comment]
Just found that some of what is needed is already done here: https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/tvplay.py

**Comment by 18928172992817182** [comment]
Just tried yt-dlp with subtitles on viafree in mpv. Found that you could do this for mpv:
```
script-opts=ytdl_hook-ytdl_path=/usr/local/bin/yt-dlp
```

It seems to find that there are subtitles but when you select one it freezes mpv completely. The only thing you get is: 

```
[ffmpeg] https: Opening 'https://vod1-hls-cdn1-vf-cdn-viaplay-tv.akamaized.net/8a4dbd8/20670544-1621951312791.ism/20670544-1621951312791-textstream_dan=1000-211.webvtt' for reading
[ffmpeg] https: Opening 'https://vod1-hls-cdn1-vf-cdn-viaplay-tv.akamaized.net/8a4dbd8/20670544-1621951312791.ism/20670544-1621951312791-textstream_swe=1000-138.webvtt' for reading
[cplayer] Video frame delayed due to waiting on subtitles.
[cplayer] Video frame delayed due to waiting on subtitles.
[cplayer] Video frame delayed due to waiting on subtitles.
[cplayer] Video frame delayed due to waiting on subtitles.
```

It is same translations. Both my solution and the manifest solution works with the libs by themselves so it is safe to assume that the issue is in mpv when it is muxing streams together.
